docs/community/20241112-node-provisioning.md

```diff
@@ -0,0 +1,88 @@
+---
+title: ClusterAPI Node Provisioning working group
```

Comment:
I'm a bit concerned about carrying over the conflation between bootstrap and provisioning into the working group...
The filename has "node provisioning" in it but the title says "node bootstrapping".

Also, I'm not sure what we mean by "cluster provisioning".

To clarify, the two concerns which are currently intertwined are the following:

1. Generating cloud-config or Ignition configuration and executing it on nodes to make OS customizations, file manipulations on disk etc.
2. Generating kubeadm configuration and executing kubeadm to turn a host into a k8s node.

I don't care too much what we call each concern as long as the two are clearly demarcated and we know which one we're discussing at any given moment.

In my WIP design document I offered to call no. 1 "provisioning" and no. 2 "bootstrap".

Reply:
Good catch, I was trying to use "cluster provisioning" and "node bootstrapping" consistently but apparently it escaped me in the title.